src/Features/CSharp/Portable/UseAutoProperty/CSharpUseAutoPropertyCodeFixProvider.cs

```diff
@@ -17,167 +16,231 @@
-        return nodeToRemove;
+        var fieldDeclaration = GetFieldDeclaration(declarator);
+        return fieldDeclaration.Declaration.Variables.Count > 1 ? declarator : fieldDeclaration;
+    }
+
+    protected override PropertyDeclarationSyntax RewriteFieldReferencesInProperty(
+        PropertyDeclarationSyntax property,
+        LightweightRenameLocations fieldLocations,
+        CancellationToken cancellationToken)
+    {
+        var propertyIdentifier = property.Identifier.WithoutTrivia();
+        var propertyIdentifierName = IdentifierName(propertyIdentifier);
+
+        var identifierNames = fieldLocations.Locations
+            .Select(loc => loc.Location.FindNode(cancellationToken) as IdentifierNameSyntax)
+            .WhereNotNull()
+            .ToSet();
+
+        var rewriter = new UseAutoPropertyRewriter(propertyIdentifierName, identifierNames);
+        return (PropertyDeclarationSyntax)rewriter.Visit(property);
     }
 
-    protected override async Task<SyntaxNode> UpdatePropertyAsync(
-        Document propertyDocument, Compilation compilation, IFieldSymbol fieldSymbol, IPropertySymbol propertySymbol,
-        PropertyDeclarationSyntax propertyDeclaration, bool isWrittenOutsideOfConstructor, CancellationToken cancellationToken)
+    protected override Task<SyntaxNode> UpdatePropertyAsync(
+        Document propertyDocument,
+        Compilation compilation,
+        IFieldSymbol fieldSymbol,
+        IPropertySymbol propertySymbol,
+        VariableDeclaratorSyntax fieldDeclarator,
+        PropertyDeclarationSyntax propertyDeclaration,
+        bool isWrittenOutsideOfConstructor,
+        bool isTrivialGetAccessor,
+        bool isTrivialSetAccessor,
+        CancellationToken cancellationToken)
     {
         var project = propertyDocument.Project;
-        var trailingTrivia = propertyDeclaration.GetTrailingTrivia();
+        var generator = SyntaxGenerator.GetGenerator(project);
 
-        var updatedProperty = propertyDeclaration.WithAccessorList(UpdateAccessorList(propertyDeclaration.AccessorList))
-                                                 .WithExpressionBody(null)
-                                                 .WithSemicolonToken(Token(SyntaxKind.None));
+        // Ensure that any attributes on the field are moved over to the property.
+        propertyDeclaration = MoveAttributes(propertyDeclaration, GetFieldDeclaration(fieldDeclarator));
 
         // We may need to add a setter if the field is written to outside of the constructor
         // of it's class.
-        if (NeedsSetter(compilation, propertyDeclaration, isWrittenOutsideOfConstructor))
+        var needsSetter = NeedsSetter(compilation, propertyDeclaration, isWrittenOutsideOfConstructor);
+        var fieldInitializer = fieldDeclarator.Initializer?.Value;
+
+        if (!isTrivialGetAccessor && !isTrivialSetAccessor && !needsSetter && fieldInitializer == null)
```

Comment:
next, we update the property to use 'accessor list' form *if needed*.

Reply:
a lot of the below code changes because it used to be able to just unilaterally assume it would convert directly to `get;` or `get;set;`.  But now we have to actually properly manipulate the individual accessors depending on if we get a simple accessor, or keep a complex one that uses `field` within it.